### PR TITLE
[Merged by Bors] - chore: backports for leanprover/lean4#4814 (part 16)

### PR DESCRIPTION
**Pull request by kim-em**
see https://github.com/leanprover-community/mathlib4/pull/15245

### CODE REVIEW

File: Mathlib/Algebra/MonoidAlgebra/Degree.lean [hunk 1]
```diff
@@ -243,7 +243,17 @@ theorem supDegree_single (a : A) (r : R) :
...
+    p a = 0 := by
+  contrapose! hlt
+  exact Finset.le_sup (Finsupp.mem_support_iff.2 hlt)
+
+theorem supDegree_withBot_some_comp {s : AddMonoidAlgebra R A} (hs : s.support.Nonempty) :
+    supDegree (WithBot.some ∘ D) s = supDegree D s := by
+  unfold AddMonoidAlgebra.supDegree
+  rw [← Finset.coe_sup' hs, Finset.sup'_eq_sup]
+
+variable [AddZeroClass A] {p q : R[A]}
```
Comment (sgouezel) [inline Mathlib/Algebra/MonoidAlgebra/Degree.lean:256]:
```suggestion
variable {p q : R[A]}
```
There is already `[AddZeroClass A]` around. I wonder why this hasn't been caught in the main branch.

**Review by sgouezel — COMMENTED** [review]
bors d+

File: Mathlib/Algebra/MonoidAlgebra/Degree.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Algebra/MonoidAlgebra/Degree.lean]: Actually, this is needed!

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**